algebra-utils/src/msm/variable_base.rs

```diff
@@ -176,11 +380,50 @@ impl VariableBaseMSM {
         G: AffineCurve,
         G::Projective: ProjectiveCurve<Affine = G>
     {
-        #[cfg(not(feature = "msm_affine"))]
-        return Self::multi_scalar_mul_mixed(bases, scalars);
 
-        #[cfg(feature = "msm_affine")]
-        return Self::multi_scalar_mul_affine(bases, scalars);   
+        let scal_len = scalars.len();
+
+        if TypeId::of::<G>() == TypeId::of::<algebra::curves::bn_382::G1Affine>()
```

Comment:
This if should be enable iff the feature "bn_382" is enabled. Add #[cfg(feature = "bn_382")] to line 385